management/branding.go

```diff
@@ -16,10 +20,106 @@ type Branding struct {
 type BrandingColors struct {
 	// Accent color
 	Primary *string `json:"primary,omitempty"`
+
 	// Page background color
+	// Only one of PageBackground and PageBackgroundGradient should be set.
+	// If both fields are set, PageBackground takes priority.
+	PageBackground *string `json:"-"`
+
+	// Page background gradient
+	// Only one of PageBackground and PageBackgroundGradient should be set.
+	// If both fields are set, PageBackground takes priority.
+	PageBackgroundGradient *BrandingPageBackgroundGradient `json:"-"`
+}
+
+type BrandingPageBackgroundGradient struct {
+	Type        *string `json:"type,omitempty"`
+	Start       *string `json:"start,omitempty"`
+	End         *string `json:"end,omitempty"`
+	AngleDegree *int    `json:"angle_deg,omitempty"`
+}
+
+type jsonPageBackgroundSolid struct {
 	PageBackground *string `json:"page_background,omitempty"`
 }
 
+type jsonPageBackgroundGradient struct {
+	PageBackgroundGradient *BrandingPageBackgroundGradient `json:"page_background,omitempty"`
+}
+
+// MarshalJSON implements the json.Marshaler interface.
+//
+// It is required to handle the json field page_background, which can either
+// be a hex color string, or an object describing a gradient.
+func (bc *BrandingColors) MarshalJSON() ([]byte, error) {
```

Comment:
Using the approach from `UserIdentity` this could look like so:

```
func (bc *BrandingColors) MarshalJSON() ([]byte, error) {

	type brandingColors BrandingColors
	type brandingColorsWrapper struct {
		*brandingColors
		RawPageBackground interface{} `json:"page_background,omitempty"`
	}

	alias := &brandingColorsWrapper{(*brandingColors)(bc), nil}

	if bc.PageBackground != nil {
		alias.RawPageBackground = bc.PageBackground
	} else if bc.PageBackgroundGradient != nil {
		alias.RawPageBackground = bc.PageBackgroundGradient
	}

	return json.Marshal(alias)
}
```

Reply:
That works perfectly. I added a check to see if both fields are `!= nil` and throw an error in that case and added credits to the commit for you.